google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -302,11 +303,16 @@ def bq_to_arrow_array(series, bq_field):
 
     field_type_upper = bq_field.field_type.upper() if bq_field.field_type else ""
 
-    if bq_field.mode.upper() == "REPEATED":
-        return pyarrow.ListArray.from_pandas(series, type=arrow_type)
-    if field_type_upper in schema._STRUCT_TYPES:
-        return pyarrow.StructArray.from_pandas(series, type=arrow_type)
-    return pyarrow.Array.from_pandas(series, type=arrow_type)
+    try:
+        if bq_field.mode.upper() == "REPEATED":
+            return pyarrow.ListArray.from_pandas(series, type=arrow_type)
+        if field_type_upper in schema._STRUCT_TYPES:
+            return pyarrow.StructArray.from_pandas(series, type=arrow_type)
+        return pyarrow.Array.from_pandas(series, type=arrow_type)
+    except ArrowTypeError:  # pragma: NO COVER
+        msg = f"""Error converting Pandas column with name: "{series.name}" to pyarrow datatype: Array, ListArray, or StructArray"""
```

Comment:
Would it be a good idea to include the type of the column that couldn't be converted?

Reply:
Done